packages/block-library/src/navigation/edit/responsive-wrapper.js

```diff
@@ -39,12 +39,20 @@ export default function ResponsiveWrapper( {
 
 	const modalId = `${ id }-modal`;
 
+	const dialogProps = {
+		className: 'wp-block-navigation__responsive-dialog',
+		...( isOpen && {
+			role: 'dialog',
+			'aria-modal': true,
+			'aria-label': __( 'Menu' ),
+		} ),
+	};
+
 	return (
 		<>
 			{ ! isOpen && (
 				<Button
 					aria-haspopup="true"
-					aria-expanded={ isOpen }
```

Comment:
Thank you for removing the `aria-expanded`attribute here, since changing the button label is enough to convey the status the modal is currently in and what will happen when the button is activated. 👍